src/main/java/teammates/common/datatransfer/QueryLogsParams.java

```diff
@@ -0,0 +1,114 @@
+package teammates.common.datatransfer;
+
+import java.time.Instant;
+
+/**
+ * Represents the parameters used for querying logs.
+ */
+public class QueryLogsParams {
+    String severityLevel;
+    String minSeverity;
+    Instant startTime;
+    Instant endTime;
+    String traceId;
+    String actionClass;
+    UserInfoParams userInfoParams;
+    String logEvent;
+    GeneralLogEntry.SourceLocation sourceLocation;
+    String exceptionClass;
+    Integer pageSize;
+    String pageToken;
+
+    public QueryLogsParams(String severityLevel, String minSeverity, Instant startTime, Instant endTime,
+            String traceId, String actionClass, String googleId, String regkey, String email, String logEvent,
+            GeneralLogEntry.SourceLocation sourceLocation, String exceptionClass, Integer pageSize, String pageToken) {
```

Comment:
This doesn't solve the previous problem of method having too many parameters being error-prone, albeit it does reduce the entrypoint of possible errors to just one.

Better way here is to use builder pattern, or at least an empty constructor with setters.

Reply:
Understood. Builder class created for query logs params.